### PR TITLE
change(test): Create test harness for calling getblocktemplate in proposal mode, but don't use it yet

### LINKED ISSUE #5685: Test the getblocktemplate RPC's response as a block template proposal in CI
## Motivation

The acceptance test could confirm that the `getblocktemplate` response in 'template' mode will be accepted if it has a valid solution by submitting the same data to the `getblocktemplate` method in 'proposal' mode.

### Design

In the acceptance test that syncs until the tip and calls the `getblocktemplate` RPC method:
- Deserialize the response
- Convert the template into a block with empty `solution` and `nonce` fields
- Hex-encode the resulting block
- Call `getblocktemplate` in 'proposal' mode
- Assert a "null" response indicating successful validation (similar to the `submitblock` test)

### PR DESCRIPTION
**Pull request by arya2**
## Motivation

We want to use `getblocktemplate` in proposal mode in the acceptance test to check that it's producing valid templates.

## Solution

- Add `proposal_block_from_template` fn for making a block proposal from `GetBlockTemplate` data
- Add a test harness for calling `getblocktemplate` in proposal mode with the hex-encoded block proposal data, but don't use it yet

## Review

Anyone can review.

### Reviewer Checklist

  - [x] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [x] Are the PR labels correct?
  - [x] Does the code do what the ticket and PR says?
    - [x] Does it change concurrent code, unsafe code, or consensus rules?
  - [x] How do you know it works? Does it have tests?

### CODE REVIEW

**Comment by teor2345** [comment]
Thanks for this, having tests makes the other PR easier to review. Would you mind rebasing this one as well?

> supersedes #5803.

I'm not sure if this PR entirely covers the tests in ticket #5803. Or do we have another ticket for testing that Zebra actually works for mining pools? 

(We can do that ticket after we've finished the other RPCs that mining pools need, but it's still a test we need to do.)

**Comment by arya2** [comment]
> Thanks for this, having tests makes the other PR easier to review. Would you mind rebasing this one as well?
> 
> > supersedes #5803.
> 
> I'm not sure if this PR entirely covers the tests in ticket #5803. Or do we have another ticket for testing that Zebra actually works for mining pools?
> 
> (We can do that ticket after we've finished the other RPCs that mining pools need, but it's still a test we need to do.)

We have https://github.com/ZcashFoundation/zebra/issues/5304, I think #5803 was supposed to be a manual version of this with the zcashd block proposal (see discussion https://github.com/ZcashFoundation/zebra/discussions/5804).

**Comment by teor2345** [comment]
> > Thanks for this, having tests makes the other PR easier to review. Would you mind rebasing this one as well?
> > > supersedes #5803.
> > 
> > 
> > I'm not sure if this PR entirely covers the tests in ticket #5803. Or do we have another ticket for testing that Zebra actually works for mining pools?
> > (We can do that ticket after we've finished the other RPCs that mining pools need, but it's still a test we need to do.)
> 
> We have #5304, I think #5803 was supposed to be a manual version of this with the zcashd block proposal (see discussion #5804).

Yep, CI is very different from a once-off test. I've updated the ticket name to make that clearer.

**Comment by teor2345** [comment]
I'm probably going to review this PR after PR #5870 gets updated, and those changes are applied here. (I think I just did my final big review on #5870.)

I also tweaked the PR description to put the "depends on" field in the mergify format:
```
Depends-On: #5870
```

https://docs.mergify.com/actions/merge/#queue-depends-on

File: zebrad/tests/common/get_block_template_rpcs/get_block_template.rs
Comment (teor2345) [inline zebrad/tests/common/get_block_template_rpcs/get_block_template.rs]: This seems like it could be a bug in the block proposal validation code or the test harness.

Can you show me one of these rejections?

File: zebra-chain/src/work/equihash.rs
Comment (teor2345) [inline zebra-chain/src/work/equihash.rs]: This default doesn't obey the consensus rules. So usually I would:
- make a named method that makes it clear it's only for a proposal
- make it test-only

File: zebrad/tests/common/get_block_template_rpcs/get_block_template.rs
Comment (teor2345) [inline zebrad/tests/common/get_block_template_rpcs/get_block_template.rs:228]: Optional: Similar to my previous comment, I'd prefer to generate this using a method with a name that includes "proposal".

But that's less important here, because it's only in test code.

File: zebrad/tests/common/get_block_template_rpcs/get_block_template.rs
Comment (teor2345) [inline zebrad/tests/common/get_block_template_rpcs/get_block_template.rs]: Using the raw local time could cause failures. The consensus rules require the `cur_time` from the block template, which has been clamped to the min and max times specified in the consensus rules.

Since we've had bugs in the time field, I'd like to test all 3 times here: `min_time`, `cur_time`, and `max_time`. All those times should produce valid blocks.

File: zebrad/tests/common/rpc_client.rs
Comment (teor2345) [inline zebrad/tests/common/rpc_client.rs]: Optional: we can say this in code so it's automatically checked:
```suggestion
    #[cfg(feature = "getblocktemplate-rpcs")]
```

File: zebrad/tests/common/get_block_template_rpcs/get_block_template.rs
Comment (teor2345) [inline zebrad/tests/common/get_block_template_rpcs/get_block_template.rs:190]: I think I'll need this function as a separate binary for PR #5944, doing it in a shell script with `jq` is difficult.

I can make these changes in my PR.

File: zebrad/tests/common/rpc_client.rs
Comment (teor2345) [inline zebrad/tests/common/rpc_client.rs]: Same here.

**Review by teor2345 — COMMENTED** [review]
Looks good!

I just bumped the previous PR so it hopefully gets merged by CI.

I think there's one consensus rule we need to change a field for, otherwise this should work well.

**Comment by teor2345** [comment]
I'm going to merge in the latest main branch so I can work off this PR, feel free to rebase or overwrite the merge if that works better for you!

**Comment by teor2345** [comment]
@mergifyio update

File: zebrad/tests/common/get_block_template_rpcs/get_block_template.rs
Comment (arya2) [reply zebrad/tests/common/get_block_template_rpcs/get_block_template.rs]: Good point/idea, updated in [`516f17c`](https://github.com/ZcashFoundation/zebra/pull/5884/commits/516f17c257e3f39f63a9fa91ec0f0d253ca02b99) to return 3 block proposals from the template using `curtime`, `mintime`, and `maxtime`.

File: zebrad/tests/common/rpc_client.rs
Comment (arya2) [reply zebrad/tests/common/rpc_client.rs]: Fixed in [`516f17c`](https://github.com/ZcashFoundation/zebra/pull/5884/commits/516f17c257e3f39f63a9fa91ec0f0d253ca02b99).

File: zebrad/tests/common/rpc_client.rs
Comment (arya2) [reply zebrad/tests/common/rpc_client.rs]: Fixed in [`516f17c`](https://github.com/ZcashFoundation/zebra/pull/5884/commits/516f17c257e3f39f63a9fa91ec0f0d253ca02b99).

File: zebra-chain/src/work/equihash.rs
Comment (arya2) [reply zebra-chain/src/work/equihash.rs]: Moved/renamed to `for_proposal` method in [`516f17c`](https://github.com/ZcashFoundation/zebra/pull/5884/commits/516f17c257e3f39f63a9fa91ec0f0d253ca02b99). 

I tried using `#[cfg(all(test, feature = "getblocktemplate-rpcs"))]` too, but rust-analyzer didn't like the `test` condition:
> no function or associated item named `for_proposal` found for struct `zebra_chain::work::equihash::Solution` in the current scope

File: zebrad/tests/common/get_block_template_rpcs/get_block_template.rs
Comment (arya2) [reply zebrad/tests/common/get_block_template_rpcs/get_block_template.rs]: I'll ~add more detailed reject-reasons~ log the error for debugging this and report back. So far, I observed one rejection from before the state request but after the merkle root validity check, and one from the merkle root validity check failing.

Update: It's been the merkle root validity in every failed test I've ran since adding the log.

<details>

```console
Child Stdout:
2023-01-13T00:52:02.161355Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("f4e06325556e4a4b98505bc82d6fa032f78c2bfda459ff536a94878b0aea6921"))})
2023-01-13T00:52:09.245160Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("c1306312951ab1031c80f179a372db66f87667c61a1181edaaeabb635f72ee59"))})
2023-01-13T00:52:13.381550Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("929a563e2b47ee62bd7d72c794f9496eee144d36948418b4fad119816d6dbe66"))})
2023-01-13T00:52:20.370420Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("8931a0aad712887ff6082b6d9b377e29279c13d407e4fa58bc83ffc60e5545cc"))})
2023-01-13T00:52:24.212879Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("08b93b573c2805acdbd54bb94e0f15b00e2fc2fcfe7c51c6d588dbc69842f335"))})
2023-01-13T00:52:24.216173Z  INFO zebra_rpc::methods::get_block_template_rpcs::get_block_template: Got error response from chain_verifier CheckProposal request verify_chain_error=Block could not be full-verified verify_chain_error.sources=[block has mismatched merkle root]
2023-01-13T00:52:29.017540Z  INFO {zebrad="d2cae99" net="Main"}:sync:try_to_sync: zebrad::components::sync: starting sync, obtaining new tips state_tip=Some(Height(1946510))
2023-01-13T00:52:35.004544Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Witnessed(WtxId { id: transaction::Hash("87b0fd3d19f219127e1bbc5e898c54cdfdd22bb08deebca824a0d1d2248cb40e"), auth_digest: AuthDigest("bce06bd569c3ae92d858575b0b4ae5b593fcabf17dc1621cd8720d4ddc51b972") })})
2023-01-13T00:52:35.020145Z  INFO {zebrad="d2cae99" net="Main"}:sync:try_to_sync: zebrad::components::sync: exhausted prospective tip set
2023-01-13T00:52:35.020188Z  INFO {zebrad="d2cae99" net="Main"}:sync: zebrad::components::sync: waiting to restart sync timeout=67s state_tip=Some(Height(1946510))
2023-01-13T00:52:39.201937Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::sync::progress: finished initial sync to chain tip, using gossiped blocks sync_percent=100.000% current_height=Height(1946510) network_upgrade=Nu5 remaining_sync_blocks=1 time_since_last_state_block=1m
2023-01-13T00:52:39.721147Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("c30d29fc68abdf32ef524ffbf544b44efd284ed8d56fd5462cbac3cee569c189"))})
2023-01-13T00:52:48.505092Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("bb508f0bec0d7ce0d986dd5eeadd5835b80bf480c80bb9ba4d2b940cac061e46"))})
2023-01-13T00:52:59.566362Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("842fd94325e0a39312ce10859241ad50d23c622ee2b97ea005de638f39ddae7a"))})
2023-01-13T00:53:09.220969Z  INFO zebra_rpc::methods::get_block_template_rpcs::get_block_template: Got error response from chain_verifier CheckProposal request verify_chain_error=Block could not be full-verified verify_chain_error.sources=[block has mismatched merkle root]
2023-01-13T00:53:11.548518Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("f5907ea28fc12c370964a31f40ebd0d602b61f30ccd10e7cd1df564721f4db3c"))})
2023-01-13T00:53:21.649750Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("d8c35d48675ec228d5f2db6f95707a3316c6f35d8bccced9c927e6f2609d92df"))})
2023-01-13T00:53:34.084137Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("7dc6f53f9f053516eadf307d1599213d1a7a2898c7c93f61e5df328e9c2a9d70"))})
2023-01-13T00:53:39.204364Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::sync::progress: finished initial sync to chain tip, using gossiped blocks sync_percent=100.000% current_height=Height(1946510) network_upgrade=Nu5 remaining_sync_blocks=2 time_since_last_state_block=2m
2023-01-13T00:53:42.021959Z  INFO {zebrad="d2cae99" net="Main"}:sync:try_to_sync: zebrad::components::sync: starting sync, obtaining new tips state_tip=Some(Height(1946510))
2023-01-13T00:53:45.563350Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("887a2e2ba058358875a3e78e99ce3a686f8c33978b67d4c4413d2bfd58a25f42"))})
2023-01-13T00:53:48.024922Z  INFO {zebrad="d2cae99" net="Main"}:sync:try_to_sync: zebrad::components::sync: exhausted prospective tip set
2023-01-13T00:53:48.024967Z  INFO {zebrad="d2cae99" net="Main"}:sync: zebrad::components::sync: waiting to restart sync timeout=67s state_tip=Some(Height(1946510))
2023-01-13T00:53:48.025045Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("cac1311390a6311483c560de127473583dddbc8be17ae2068d12ba35da6f5276"))})
2023-01-13T00:53:54.226556Z  INFO zebra_rpc::methods::get_block_template_rpcs::get_block_template: Got error response from chain_verifier CheckProposal request verify_chain_error=Block could not be full-verified verify_chain_error.sources=[block has mismatched merkle root]
2023-01-13T00:53:55.430270Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("7ec85629292c5ee502055cccf128ce93043dd05c3fb73054ee8ba5f1cc25392d"))})
2023-01-13T00:54:03.181670Z  INFO {zebrad="d2cae99" net="Main"}:{peer=Out("120.24.79.67:8233")}:msg_as_req{msg="inv"}:inbound:download_and_verify{hash=00000000009a66789c1f53b29d88ed7b53ae5da79e81fcef0ead0693a178bac9}: zebrad::components::inbound::downloads: downloaded and verified gossiped block height=Height(1946511)
2023-01-13T00:54:08.294108Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("c6c69e2e9f77cd34f46f916f9c0ef6f939920b87d965f181cc7d7d1589af928b"))})
2023-01-13T00:54:19.990629Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("0e9736bfccd97a799d12d844252650109c780d487a172a4e97bb3049a76c3e5d"))})
2023-01-13T00:54:22.252819Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("d289b739393334b564d67e94b526a612fde5759c6ed6077edfb2ea9b57db7b65"))})
2023-01-13T00:54:31.721513Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("97a0e5fc3edb740b53693517cb71727059bc77d7fcafc83af4629e033472185f"))})
2023-01-13T00:54:39.205544Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::sync::progress: finished initial sync to chain tip, using gossiped blocks sync_percent=100.000% current_height=Height(1946511) network_upgrade=Nu5 remaining_sync_blocks=0 time_since_last_state_block=0s
2023-01-13T00:54:39.229069Z  INFO zebra_rpc::methods::get_block_template_rpcs::get_block_template: Got error response from chain_verifier CheckProposal request verify_chain_error=Block could not be full-verified verify_chain_error.sources=[block has mismatched merkle root]
2023-01-13T00:54:42.115582Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("c68189da526633822d24ad0bd25f18dc99e8ac3ddb5dce6125880eefae77c0ff"))})
2023-01-13T00:54:46.122321Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Witnessed(WtxId { id: transaction::Hash("5dd8fb2eeff56408313ef791d7947d030f69cb164f3a630554b4dc7230e03de8"), auth_digest: AuthDigest("6e9740005109d197fe0f096b6b6d2a387ac831264d32e826a67a9f56714ee2c5") })})
2023-01-13T00:54:53.927225Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("8a46a8d549cb08e4fbbbcda648ba4b6714b8bdb477158c1ba09da3e214d3732f"))})
2023-01-13T00:54:55.026407Z  INFO {zebrad="d2cae99" net="Main"}:sync:try_to_sync: zebrad::components::sync: starting sync, obtaining new tips state_tip=Some(Height(1946511))
2023-01-13T00:55:01.029520Z  INFO {zebrad="d2cae99" net="Main"}:sync:try_to_sync: zebrad::components::sync: exhausted prospective tip set
2023-01-13T00:55:01.029606Z  INFO {zebrad="d2cae99" net="Main"}:sync: zebrad::components::sync: waiting to restart sync timeout=67s state_tip=Some(Height(1946511))
2023-01-13T00:55:04.526417Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("07860f8230cc492a3dfca1e1e63eef7c8b7f8387e361ab974a515854fe0dfe2b"))})
2023-01-13T00:55:05.072360Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("d2307f2d9f29484886af12ded6cd74660e83902c1fa51de55ede1bd72b33cfa8"))})
2023-01-13T00:55:14.251231Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("1ba29ef7ddca1c31db26ac7fd4d2e0ed3d6875bc6667c25df156170eec2e54ee"))})
2023-01-13T00:55:24.232503Z  INFO zebra_rpc::methods::get_block_template_rpcs::get_block_template: Got error response from chain_verifier CheckProposal request verify_chain_error=Block could not be full-verified verify_chain_error.sources=[block has mismatched merkle root]
2023-01-13T00:55:30.627330Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::mempool::gossip: sending mempool transaction broadcast request=AdvertiseTransactionIds({Legacy(transaction::Hash("7611237e817d3d35b692d11fc93c6b5e72b1c6fc1d8f87a41b73a4fcc3b3d764"))})
2023-01-13T00:55:39.207396Z  INFO {zebrad="d2cae99" net="Main"}: zebrad::components::sync::progress: finished initial sync to chain tip, using gossiped blocks sync_percent=100.000% current_height=Height(1946511) network_upgrade=Nu5 remaining_sync_blocks=1 time_since_last_state_block=1m
2023-01-13T00:55:50.877738Z  INFO {zebrad="d2cae99" net="Main"}:{peer=Out("35.72.109.227:8233")}:msg_as_req{msg="inv"}:inbound:download_and_verify{hash=0000000000c296d571f7ec62da09010a69068d2b766a550e37866a4f169532ab}: zebrad::components::inbound::downloads: downloaded and verified gossiped block height=Height(1946512)
2023-01-13T00:56:08.030459Z  INFO {zebrad="d2cae99" net="Main"}:sync:try_to_sync: zebrad::components::sync: starting sync, obtaining new tips state_tip=Some(Height(1946512))
2023-01-13T00:56:08.511621Z  INFO {zebrad="d2cae99" net="Main"}:sync:try_to_sync: zebrad::components::sync: exhausted prospective tip set
2023-01-13T00:56:08.511646Z  INFO {zebrad="d2cae99" net="Main"}:sync: zebrad::components::sync: waiting to restart sync timeout=67s state_tip=Some(Height(1946512))
2023-01-13T00:56:09.235425Z  INFO zebra_rpc::methods::get_block_template_rpcs::get_block_template: Got error response from chain_verifier CheckProposal request verify_chain_error=Block could not be full-verified verify_chain_error.sources=[block has mismatched merkle root]
```
</details>

Same story in CI: https://github.com/ZcashFoundation/zebra/actions/runs/3907156559/jobs/6676518283#step:8:637

Did the `zcashd` proposal log anything interesting?

**Comment by teor2345** [comment]
I think that running this new test in CI is blocked on fixing the merkle tree bug in the `getblocktemplate` RPC.

File: zebrad/tests/common/get_block_template_rpcs/get_block_template.rs
Comment (arya2) [reply zebrad/tests/common/get_block_template_rpcs/get_block_template.rs]: The transactions weren't sorted when generating the default Merkle root 🤷

File: zebra-rpc/src/methods/get_block_template_rpcs/types/get_block_template.rs
Comment (teor2345) [inline zebra-rpc/src/methods/get_block_template_rpcs/types/get_block_template.rs:268]: Just a note for next time: please try to avoid large changes in a PR when someone has based another PR off it. It makes merging any changes into their PR harder.

There's also no need to backport changes that will happen in the next PR.

File: zebrad/tests/common/rpc_client.rs
Comment (teor2345) [inline zebrad/tests/common/rpc_client.rs]: I'll push a commit that fixes these clippy warnings, then approve this PR.

**Comment by teor2345** [comment]
This PR can't merge for now:
> I don't think we should do this in CI until we can confirm it manually

https://github.com/ZcashFoundation/zebra/issues/5685#issuecomment-1381915157

File: zebrad/tests/common/rpc_client.rs
Comment (teor2345) [reply zebrad/tests/common/rpc_client.rs]: I pushed commit 0702797, but I can't approve because I'm the last committer.

**Comment by teor2345** [comment]
> This PR can't merge for now:
> 
> > I don't think we should do this in CI until we can confirm it manually
> 
> [#5685 (comment)](https://github.com/ZcashFoundation/zebra/issues/5685#issuecomment-1381915157)

I unblocked this PR by disabling the test for now, and updating the PR description.

So we're ok to merge the test harness, and all the PRs that depend on it, because it doesn't actually change CI yet.

**Comment by teor2345** [comment]
I can't work out how to stop it closing issue #5685 though!

File: zebra-rpc/src/methods/get_block_template_rpcs/types/get_block_template.rs
Comment (arya2) [reply zebra-rpc/src/methods/get_block_template_rpcs/types/get_block_template.rs:268]: Noted. I should've checked if those files had been changed in your PR, my bad here.

File: zebrad/tests/common/rpc_client.rs
Comment (arya2) [reply zebrad/tests/common/rpc_client.rs]: > I'll push a commit that fixes these clippy warnings, then approve this PR.

Thank you!

> I pushed commit https://github.com/ZcashFoundation/zebra/commit/070279742ef548f3ef2a6e04dfdc1b9e28ff821a, but I can't approve because I'm the last committer.

Hm. I'll redo your commits so you can approve.

**Review by teor2345 — APPROVED** [review]

**Comment by teor2345** [comment]
Failed due to a GitHub runner network or setup issue in the merge queue:
https://github.com/ZcashFoundation/zebra/actions/runs/3934132575

**Comment by teor2345** [comment]
@mergifyio refresh

**Pull request merged**
